pkg/monitors/vsphere/runner.go

```diff
@@ -36,8 +34,7 @@ func (r *runner) run() {
 		r.log.WithError(err).Error("firstTimeSetup failed")
 		return
 	}
-	dps := r.vsm.retrieveDatapoints()
-	r.monitor.Output.SendDatapoints(dps...)
+	r.vsm.retrieveDatapoints()
```

Comment:
Call this something like `r.vsm.Run()` or something like that.  It isn't really retrieving datapoints, it is doing a bunch of stuff to generate them.

Reply:
The main thing is that this function is called entierly for its side effect so its name should reflect that.

Reply:
Agree, but couldn't think of a name so I had left it. Changed it to `generateDatapoints`